src/PrestaShopBundle/Controller/Admin/Sell/Catalog/FeatureController.php

```diff
@@ -231,63 +222,55 @@ private function setListActions(HelperListConfiguration $helperListConfiguration
     {
         $controllerConfiguration = $this->getControllerConfiguration();
 
-        $this->addActionList(new ListHeaderToolbarAction('new', [
-            //@todo: replace by $this->generateUrl('admin_features_add') when creation is fully migrated
-            //@todo: can i generate link without using controller config?
-            'href' => $controllerConfiguration->legacyCurrentIndex . '&addfeature&token=' . $controllerConfiguration->token,
-            'desc' => $this->trans('Add new', 'Admin.Actions'),
-        ]), $helperListConfiguration);
-
-        $this->addActionList(new ListRowAction('view'), $helperListConfiguration);
-        $this->addActionList(new ListRowAction('edit'), $helperListConfiguration);
-        $this->addActionList(new ListRowAction('delete'), $helperListConfiguration);
-
-        $this->addActionList(new ListBulkAction('delete', [
-            'text' => $this->trans('Delete selected', 'Admin.Actions'),
-            'icon' => 'icon-trash',
-            'confirm' => $this->trans('Delete selected items?', 'Admin.Notifications.Warning'),
-        ]), $helperListConfiguration);
+        $helperListConfiguration
+            ->addRowAction('view')
+            ->addRowAction('edit')
+            ->addRowAction('delete')
+            ->addToolbarAction('new', [
+                //@todo: replace by $this->generateUrl('admin_features_add') when creation is fully migrated
+                //@todo: can i generate link without using controller config?
+                'href' => $controllerConfiguration->legacyCurrentIndex . '&addfeature&token=' . $controllerConfiguration->token,
+                'desc' => $this->trans('Add new', 'Admin.Actions'),
+            ])
+            ->addBulkAction('delete', [
+                'text' => $this->trans('Delete selected', 'Admin.Actions'),
+                'icon' => 'icon-trash',
+                'confirm' => $this->trans('Delete selected items?', 'Admin.Notifications.Warning'),
+            ])
+        ;
     }
 
     /**
-     * Define fields in the list.
-     *
-     * @return void
+     * @param HelperListConfiguration $helperListConfiguration
      */
     private function setListFields(HelperListConfiguration $helperListConfiguration): void
     {
-        $this->addListField(new Field(
-            'id_feature', [
-                'title' => $this->trans('ID', 'Admin.Global', []),
+        $helperListConfiguration->fieldsList = [
+            'id_feature' => [
```

Comment:
as discussed, this change aims to make it easier to copy-paste code from legacy instead of initiating those new classes (which wasn't used anywhere else actually)

Reply:
This will please @eternoendless ^^